cmd/crates/soroban-test/tests/it/integration/ledger/entry.rs

```diff
@@ -0,0 +1,553 @@
+
+    let data_entry = &parsed.entries[0];
+    let name_bounded_string = StringM::<64>::try_from(data_name).unwrap();
+    let expected_data_key = LedgerKey::Data(LedgerKeyData {
+        account_id,
+        data_name: String64::from(name_bounded_string),
+    });
+    assert_eq!(data_entry.key, expected_data_key);
+    assert!(matches!(data_entry.val, LedgerEntryData::Data { .. }));
+}
+
+// contract data tests
+#[tokio::test]
+async fn ledger_entry_contract_data() {
+    let sandbox = &TestEnv::new();
+    let test_account_alias = "test";
+    let contract_id = deploy_contract(
+        sandbox,
+        HELLO_WORLD,
+        DeployOptions {
+            deployer: Some(test_account_alias.to_string()),
+            ..Default::default()
+        },
+    )
+    .await;
+
+    let storage_key = "COUNTER";
+    let storage_key_xdr = ScVal::Symbol(storage_key.try_into().unwrap())
+        .to_xdr_base64(Limits::none())
+        .unwrap();
+
+    // update contract storage
+    sandbox
+        .invoke_with_test(&["--id", &contract_id, "--", "inc"])
+        .await
+        .unwrap();
+
+    // get entry by key
+    let key_output = sandbox
+        .new_assert_cmd("ledger")
+        .arg("entry")
+        .arg("fetch")
+        .arg("contract")
+        .arg(&contract_id)
+        .arg("--network")
+        .arg("testnet")
+        .arg("--key")
+        .arg(storage_key)
+        .assert()
+        .success()
+        .stdout_as_str();
+    let parsed_key_output: FullLedgerEntries =
+        serde_json::from_str(&key_output).expect("Failed to parse JSON");
+    assert!(!parsed_key_output.entries.is_empty());
+
+    // get entry by key xdr
+    let key_xdr_output = sandbox
+        .new_assert_cmd("ledger")
+        .arg("entry")
+        .arg("fetch")
+        .arg("contract")
+        .arg(&contract_id)
+        .arg("--network")
+        .arg("testnet")
+        .arg("--key-xdr")
+        .arg(storage_key_xdr)
+        .assert()
+        .success()
+        .stdout_as_str();
+
+    let parsed_key_xdr_output: FullLedgerEntries =
+        serde_json::from_str(&key_xdr_output).expect("Failed to parse JSON");
+    assert!(!parsed_key_xdr_output.entries.is_empty());
+
+    let expected_contract_data_key = expected_contract_ledger_key(&contract_id, storage_key).await;
+
+    assert_eq!(parsed_key_output.entries[0].key, expected_contract_data_key);
+    assert!(matches!(
+        parsed_key_output.entries[0].val,
+        LedgerEntryData::ContractData { .. }
+    ));
+
+    assert_eq!(
+        parsed_key_xdr_output.entries[0].key,
+        expected_contract_data_key
+    );
+    assert!(matches!(
+        parsed_key_xdr_output.entries[0].val,
+        LedgerEntryData::ContractData { .. }
+    ));
+
+    // the output should be the same regardless of key format
+    assert_eq!(parsed_key_output.entries, parsed_key_xdr_output.entries);
+}
+
+// top level test
+// todo: test --claimable-id, --pool-id,
```

Comment:
similar question as with --offer -- how can i add claimable balances and pools in test chain in order to test this?

Reply:
@elizabethengelman you may need to create claimable balances and pools on testnet
- claimable balances: https://quest.stellar.org/learn/series/3/quest/3
- pools: https://quest.stellar.org/learn/series/3/quest/5

Reply:
Thanks for these links! I was able to create the these on the network that we're using in the gh runner 👌